data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala

```diff
@@ -186,18 +184,19 @@ trait PlayWsDownloader {
       }
     }
 
-    processDownloads(message, toDownload, processItem, parallelism)
+    processDownloads(message, toDownload, processItem)
   }
 
   private def processDownloads[T, R](
       message: String,
       toDownload: Set[T],
-      processItem: (AhcWSClient, T, ProgressBar) => Future[R],
-      parallelism: Int
+      processItem: (AhcWSClient, T, ProgressBar) => Future[R]
   ): Seq[R] = {
 
     def processItems(client: AhcWSClient, progress: ProgressBar) = {
-
+      // use minimal concurrency to avoid abuse rate limit error which is triggered
+      // by making too many calls in a short period of time, see https://github.com/scalacenter/scaladex/issues/431
+      val parallelism = 1
       Source(toDownload).mapAsyncUnordered(parallelism) { item =>
```

Comment:
[Done](https://github.com/scalacenter/scaladex/pull/470/files#diff-9b0b9f250047a450506bda2259c11537R203)

And Github didn't give any stats about the rate limit and abuse rate limits are triggered since if they shared that info people could use it to work around them. I tested with higher concurrency and got the abuse rate limit errors so I think `parallelism = 4` is good enough